rag/advanced_rag/harness/action_session.py

```
@@ -0,0 +1,1259 @@
+    from rag.advanced_rag.harness.arithmetic import compute_from_facts
+
+    from rag.advanced_rag.harness.tools.search import _base_chat_mdl
+
+    question = str(args.get("question") or "")
+    facts = [str(f) for f in (args.get("facts") or []) if str(f).strip()]
+    mdl = _base_chat_mdl(tools)
+    if mdl is None:
+        return [{"kind": "calculate", "error": "no model"}], []
+    try:
+        res = await compute_from_facts(mdl, question, facts)
+    except Exception:
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)